udata/migrations/2020-06-11-add-resource-fs-filename.py

```diff
@@ -0,0 +1,111 @@
+'''
+The purpose here is to fill every resource with a fs_filename string field.
+The migration iterates on every file in the ressource storage.
+It extracts the dataset's slug from the file path, retrieves the dataset,
+and iterates on the dataset's resources to find the one owning the file.
+If one is found the fs_filename field is filled.
+If none is found the file is deleted.
+'''
+import logging
+
+from udata.core import storages
+from udata.models import Dataset, CommunityResource, Organization, User, Reuse
+
+log = logging.getLogger(__name__)
+
+
+def migrate(db):
+    log.info('Processing resources and community resources.')
+
+    resource_files = list(storages.resources.list_files())
+    avatar_files = list(storages.avatars.list_files())
+    image_files = list(storages.images.list_files())
+
+    resource_deletion_count = 0
+    avatar_deletion_count = 0
+    image_deletion_count = 0
+
+    for fs_filename in resource_files:
+        split_str = fs_filename.split('/')
+        dataset_slug = split_str[0]
+        dataset = Dataset.objects(slug=dataset_slug).first()
```

Comment:
I think we have a problem here: what if the dataset's slug has changed? we will delete resources associated with the old slug that maybe should still be accessible.

We could use the SlugFollow mechanism but I don't think it's enough, it hasn't always been there as some slugs can be reused over time.

So... Maybe we should build a list of all the URLs of the resources and community resources hosted on data.gouv.fr (`filetype = file` should help and diff it w/ the `resources_list`.